feat: Move versioned feature history into feature details modal

kyle-ssg opened the pull request:
Thanks for submitting a PR! Please check the boxes below:

- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Moves the feature history page to within the feature modal

![image](https://github.com/user-attachments/assets/d25376a2-9711-40d0-9cca-8043cebcf979)


## How did you test this code?

E2E tests need to be changed however no functionality has changed except removing the feature select.

rolodato commented:
Some feedback:

* The contents of the History tab can expand beyond what I'm allowed to scroll to
* The contents of each entry's Quick Compare has too much whitespace for simple changes like turning a flag on/off
* The "View History" option in the Features list links to the original History page, which is now a 404
* The details view for each history entry looks orphaned, like it doesn't belong to any sidebar category (Features/Scheduling/Change Requests/etc). I wonder if there's a good way to permalink history entry details inside the modal as well, or if it would make sense to hide the sidebar in cases like these where it doesn't make sense. For example, viewing a single commit in GitHub doesn't feel out of place and has all sorts of links to browse to related items (branch, pull request, commit's files etc): https://github.com/Flagsmith/flagsmith/commit/f9cc1b0f1ca81182377efa5a755bc18d6087a595

https://github.com/user-attachments/assets/9737c8ff-096c-4d44-a02d-a32aa2323fb6

kyle-ssg commented:
> * The contents of the History tab can expand beyond what I'm allowed to scroll to
Fixed
> * The contents of each entry's Quick Compare has too much whitespace for simple changes like turning a flag on/off
Fixed
> * The "View History" option in the Features list links to the original History page, which is now a 404
Adjusted to link to tab within modal
> * The details view for each history entry looks orphaned, like it doesn't belong to any sidebar category 
Added breadcrumb to feature

novakzaballa commented on frontend/web/components/FeatureHistory.tsx:
```diff
@@ -56,57 +39,37 @@ const FeatureHistoryPage: FC<FeatureHistoryPageType> = ({ match, router }) => {
-            theme={'page'}
-          />
-        )}
+        {/*{!!versionLimit && (*/}
```
Is there any reason to keep this commented?

kyle-ssg replied:
Yes, because we're adjusting the version limit to be time based and I want to keep this here

kyle-ssg replied:
It'll be updated next week

novakzaballa reviewed (commented):
The code looks good, I just left a small comment.

kyle-ssg commented:
@rolodato final review ? 🙏

rolodato reviewed (approved):
Suggest renaming this PR to `feat: Move versioned feature history into feature details modal` before merging
